core/domain/exp_services.py

```diff
@@ -1622,6 +1623,43 @@ def get_next_page_of_all_non_private_commits(
     ) for entry in results], new_urlsafe_start_cursor, more)
 
 
+def get_image_filenames_from_exploration(exploration):
+    """ Get the image filenames from the exploration.
+
+    Args:
+        exploration: object. The exploration itself.
+
+    Returns:
+       list. The list containing the name of the image files in the exploration
+    """
+    filenames = []
+    for state in exploration.states.itervalues():
+        if state.interaction.id == 'ImageClickInput':
+            if isinstance(state.interaction.customization_args['imageAndRegions']['value']['imagePath'], dict): # pylint: disable=line-too-long
```

Comment:
Why wouldn't it be a dict? Shouldn't it be one or the other?

Don't hedge unless you have a specific reason to do so (e.g. you _know_ that there are two possible cases that currently arise in production -- but in that case there should really be a comment here explaining how you know this). Otherwise, code for "correct" case, but throw/yield an error if you get something that violates your assumption.

Reply:
Hi @vojtechjelinek , I think this can be removed right?. Because the PR involving saving `filepath` attribute as dict wasn't there in the previous release. We just added it for safety so that even if the new explorations have the `filepath` attribute as dict or string it handles both the cases.
What do you say @vojtechjelinek ?

Reply:
Yeah, all the image components in productions should be just string.

Reply:
OK, I think we have a problem. Why is there a PR saving the filepath as a dict? Has this been released yet?

This is really not the way to do it, you'll end up having some filepaths as dicts and some as strings. The customization arg values need to be deterministic. We need to block this change from going out otherwise it is going to mess up production.

@nithusha21

Reply:
We need the filepath to contain the dimensions of the image so we are able to reserve the space for the image and show laoding indicator in that place. The PR is not in production, it is in the development branch. 

The PR must be added to production together with the second one-off job that would modify the previous iamge abojects and adds the dimensions to them.

Reply:
OK. But that's via an exploration migration right? So at any given time, _all_ explorations in prod will have a single format?

In that case why is there an if statement here? It should be one or the other.

Reply:
Since there the filepath is only string as of now, I have removed the if condition. Done!

Reply:
Just saw the ping! I don't remember any issues while testing images. Not sure what impact on the end user this change could have done. 

However I think it would be better (if possible) to have a flag that can be flipped post migration to the new structure to use the dict form instead of string. This way the release coordinator can have more control and also we have don't have to revert anything in case the migration fails. Else in case we do release your PR that makes changes to the structure and the migration fails then we could be in trouble.

Maybe what I said is out of context as I am not familiar with the code changes you are doing. If so please let me know :)

Reply:
Oh ok. Thanks for the info 👍